pkl-commons/src/main/kotlin/org/pkl/commons/annotations/PklExperimental.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright © 2024-2025 Apple Inc. and the Pkl project authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pkl.commons.annotations
+
+/**
+ * Marks experimental APIs in Kotlin or Java. Such APIs may be publicly exposed by the Pkl project,
+ * but are not provided with a stability guarantee, and require opt-in use from Kotlin.
+ */
+@RequiresOptIn(level = RequiresOptIn.Level.WARNING)
+@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.BINARY)
+annotation class PklExperimental
```

Comment:
this annotation is added as an escape hatch for public API pins. it can be used like this:
```kotlin
@PklExperimental
public class SomeClass {
  // ...
}
```
Then, in user code:
```kotlin
@OptIn(PklExperimental::class)
fun someUserFunction {
  // usage of an experimental type
  val x = SomeClass()
}
```
Omission of the `@OptIn` generates a compile-time warning, and any type annotated with `@PklExperimental` is omitted from API pins.